src/test/java/seedu/address/model/UserPrefsTest.java

```diff
@@ -18,4 +24,57 @@ public void setAddressBookFilePath_nullPath_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> userPrefs.setAddressBookFilePath(null));
     }
 
+    @Test
+    public void equals() {
+        UserPrefs testUserPref = new UserPrefs();
+
+        // same values -> returns true
+        UserPrefs userPrefsCopy = new UserPrefs(testUserPref);
+        assertEquals(testUserPref, userPrefsCopy);
+
+        // same object -> returns true
+        assertEquals(testUserPref, testUserPref);
+
+        // null -> returns false
+        assertNotEquals(testUserPref, null);
+
+        // different type -> returns false
+        assertNotEquals(testUserPref, 5);
+
+        // different guiSettings -> returns false
+        UserPrefs differentUserPrefs1 = new UserPrefs();
+        differentUserPrefs1.setGuiSettings(new GuiSettings(1, 2, 3, 4));
+        assertNotEquals(differentUserPrefs1, testUserPref);
+
+        // different addressBookFilePath -> returns false
+        UserPrefs differentUserPrefs2 = new UserPrefs();
+        differentUserPrefs2.setAddressBookFilePath(Paths.get("data" , "test.json"));
+        assertNotEquals(differentUserPrefs2, testUserPref);
+
+        // different projectsFolderFilePath -> returns false
+        UserPrefs differentUserPrefs3 = new UserPrefs();
+        differentUserPrefs3.setProjectsFolderFilePath(Paths.get("data" , "test.json"));
+        assertNotEquals(differentUserPrefs3, testUserPref);
+
+    }
+
+    @Test
+    public void hashCode_success() {
+        UserPrefs userPref1 = new UserPrefs();
+        UserPrefs userPref2 = new UserPrefs();
+        userPref2.setGuiSettings(new GuiSettings(1, 2, 3, 4));
+        UserPrefs userPref3 = new UserPrefs();
+        userPref3.setAddressBookFilePath(Paths.get("data" , "test.json"));
+        UserPrefs userPref4 = new UserPrefs();
+        userPref4.setProjectsFolderFilePath(Paths.get("data" , "test.json"));
+        int hashcode1 = userPref1.hashCode();
+        int hashcode2 = userPref2.hashCode();
+        int hashcode3 = userPref3.hashCode();
+        int hashcode4 = userPref4.hashCode();
+        assertEquals(hashcode1, userPref1.hashCode());
```

Comment:
I think it would be better to add a check whether two objects which are equal have the same hashcode. This fulfils the second point of the hashcode contract as stated in https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode().

Also fyi it's not really necessary to check that different objects have different hashcodes (not part of the hashcode contract) but i think it's ok to leave it in this case!